qemu/cpus.c

```diff
@@ -191,15 +169,10 @@ static int qemu_tcg_init_vcpu(CPUState *cpu)
         uc->tcg_halt_cond = cpu->halt_cond;
         snprintf(thread_name, VCPU_THREAD_NAME_SIZE, "CPU %d/TCG",
                 cpu->cpu_index);
-        if (qemu_thread_create(uc, cpu->thread, thread_name, qemu_tcg_cpu_thread_fn,
-                cpu, QEMU_THREAD_JOINABLE))
-            return -1;
+        qemu_thread_get_self(uc, cpu->thread);
```

Comment:
QEMU kinda supports multiple CPUs multiplexed onto a single thread, where it alternates calling the tcg exec functions https://github.com/lunixbochs/unicorn/blob/master/qemu/cpus.c#L200

Some of this might be lost in translation to Unicorn. It honestly might make the most sense to eventually remove all support for multiple threads and CPUs in Unicorn and expose any necessary sync primitives to the API, but I think we need a more clear use case and better idea of the upsides and downsides before someone works on it.